### ISSUE TITLE
extract text from the --save-crop images yolov5

### ISSUE THREAD

**Issue by bachimanchiajay** [ultralytics/yolov5#9839]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

after executing the detect.py file --save-crop i can able to crop the class images but i need the text as well before image writing.
for eg : i tested with one image having text model predicted the text classes. i want to extract the text in the predicted coordinates before copping the class images

**Comment by glenn-jocher** [comment]
@bachimanchiajay 👋 Hello! Thanks for asking about **cropping results** with YOLOv5 🚀.  Cropping bounding box detections can be useful for training classification models on box contents for example. This feature was added in PR https://github.com/ultralytics/yolov5/pull/2827. You can crop detections using either detect.py or YOLOv5 PyTorch Hub:

## detect.py
Crops will be saved under `runs/detect/exp/crops`, with a directory for each class detected.
```bash
python detect.py --save-crop
```
<img width="450" alt="Original" src="https://user-images.githubusercontent.com/26833433/139577918-65cc8f5d-741d-4a88-8753-0154013c624d.jpg">

<img width="1067" alt="Crop" src="https://user-images.githubusercontent.com/26833433/139577896-aa26b07c-466d-4057-9bfe-b01b56007587.png">

## YOLOv5 PyTorch Hub
Crops will be saved under `runs/detect/exp/crops` if `save=True`, and also returned as a dictionary with crops as numpy arrays.
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, custom

# Images
img = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results
crops = results.crop(save=True) 
# -- or --
crops = results.crop(save=True, save_dir='runs/detect/exp')  # specify save dir
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by bachimanchiajay** [comment]
i want to extract the text from the predicted results, have to pass the predicted bounding box coordinates to the textract can you help me in that

**Comment by bachimanchiajay** [comment]
### 
import cv2
import numpy as np
import os
import pytesseract as pt
import torch
from PIL import Image
import matplotlib.pyplot as plt
model = torch.hub.load('/home/pkumar/yolov5/', 'custom', path='/home/pkumar/Desktop/yolo_dataentry_models/best.pt',source='local')
def get_text(image_path, threshold):
    img = cv2.imread(image_path)
    Y, X,d = img.shape
    result = model(img)
    df = result.pandas().xyxy[0]
    df = df[df['confidence']>=threshold]
    df = df.drop(['class', 'name','confidence'], axis=1)
    print(df.head())
    output = []
    i = 0
    values = df.values.astype(int)
    for xmin,ymin,xmax,ymax in df.values.astype(int):
        xmin = max(0, xmin-3)
        ymin = max(0, ymin-3)
        xmax = min(X, xmax+3)
        ymax = min(Y, ymax+3)
        cv2.rectangle(img, (xmin,ymin),(xmax,ymax),(0,255,0),3)
        crop_img = img[ymin:ymax, xmin:xmax]
        crop_img = cv2.cvtColor(crop_img, cv2.COLOR_BGR2GRAY)
        crop_img = cv2.resize(crop_img, None, fx=4,fy=4, interpolation=cv2.INTER_CUBIC)
        text = pt.image_to_string(crop_img).strip()
    output.append(text)
    return output,img
path = '/home/pkumar/Downloads/UP_images/SAO1_0deblur.png'
text,image= get_text(path, 0.5)`

**Comment by bachimanchiajay** [comment]
This one is not extracting all the classes only getting two class values  @glenn-jocher  can you cross check the above method once

**Comment by Aayan37** [comment]
@glenn-jocher please can you let me know the code for YoloV5 with bounding box cropped saved in some location. I have tried integrating method https://github.com/ultralytics/yolov5/pull/2827! but its not working.. can you please share the latest code.

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about **cropping results** with YOLOv5 🚀.  Cropping bounding box detections can be useful for training classification models on box contents for example. This feature was added in PR https://github.com/ultralytics/yolov5/pull/2827. You can crop detections using either detect.py or YOLOv5 PyTorch Hub:

## detect.py
Crops will be saved under `runs/detect/exp/crops`, with a directory for each class detected.
```bash
python detect.py --save-crop
```
<img width="450" alt="Original" src="https://user-images.githubusercontent.com/26833433/139577918-65cc8f5d-741d-4a88-8753-0154013c624d.jpg">

<img width="1067" alt="Crop" src="https://user-images.githubusercontent.com/26833433/139577896-aa26b07c-466d-4057-9bfe-b01b56007587.png">

## YOLOv5 PyTorch Hub
Crops will be saved under `runs/detect/exp/crops` if `save=True`, and also returned as a dictionary with crops as numpy arrays.
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, custom

# Images
img = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results
crops = results.crop(save=True) 
# -- or --
crops = results.crop(save=True, save_dir='runs/detect/exp')  # specify save dir
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by Aayan37** [comment]
where can i find the plot_one_box function? I couldnt find it in utils.plots. Only save_one_box is present in that file.

**Comment by Aayan37** [comment]
@glenn-jocher I have tried this but its not cropping the bounding boxes.  can you please share the complete detect.py with cropping images code present in it. I have tried Abdurrahman, @burhr2 defined codes as well. May be I'm missing something, can you please share the detailed code for that. Thanks in Advance!!

**Comment by glenn-jocher** [comment]
@Aayan37 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃
